doc/Frontend-Documentation.md

```diff
@@ -0,0 +1,628 @@
+# Packit App
+
+## Overview
+
+**What does it do?**
+
+**Packit** is a web application designed to provide users with a comprehensive link shortening service. This powerful tool offers several key functionalities, making it a versatile solution for personal and professional use.
+
+Packit allows users to:
+- **Shorten Links**: Quickly convert long URLs into shorter, more manageable links.
+- **Customize Shortened Links**: Personalize the shortened links to make them more memorable and relevant.
+- **Generate QR Codes**: Create QR codes for the shortened links, facilitating easy sharing and access via mobile devices.
+- **Analyze Link Performance**: Access both detailed and general analytics about the shortened links, including metrics such as click counts, geographic data, and referral sources.
+
+Packit is designed to be user-friendly, enabling users to manage their links efficiently and gain valuable insights into their link traffic.
+
+## Functionality of the Project
+
+### Link Shortening
+Packit offers a straightforward process for shortening long URLs. Users simply paste their lengthy URLs into the application, and with a single click, Packit generates a shortened link. This shortened link retains the original destination but is significantly reduced in character count, making it easier to share and manage.
```

Comment:
into the application, and with a single  when you use a "and", no need to put a comma before